proposals/eni.md

```diff
@@ -0,0 +1,129 @@
+                                            |   +------------------+ |   |
+                                          +----+|default via eth1  | +   |
+                             172.31.13.14 |eth1||169.254.170.2 via | |   |
+                                          +----+|     dev ve_en1   | +   |
+                                            |   +------------------+ |---+
+                                            |       169.254.x.2      +
+                                            +-----------+------------+
+                                                        |ve_en1
+                                                        |
+                                                        |
+                                                        |
+                                                        |
+                          +-------------------------------------------------------------+
+                          |default(global) namespace    |                               |
+                          |                             |                               |
+                          |                     +-------+-------+       +-------------+ |  +------------+
+                          |                     |    ve_br_en1  |       |             | |  |net="bridge"|
+             172.31.0.226 |   +-----------+     |               |       |             +-|--|container   |
+                       +----+ |           |     |ecs-eni bridge |       |             | |  +------------+
+                       |eth0| | ECS Agent |     |               |       | docker0     | |
+                       +----+ +-----------+     |               |       | bridge      | |
+                          |                     |               |       |             | |
+                          |                     |  169.254.x.1  |       |             | |
+                          |                     +---------------+       +-------------+ |
+                          |                                                             |
+                          +-------------------------------------------------------------+
+
+In this diagram, the primary network interface is still being used on the instance
+by the `docker0` bridge to let containers launched with `net=bridge` to communicate
+with entities outside the instance. A secondary network interface has also been
+attached for the purpose of being used by a task (`eth1`). This interface has been
+moved to the `ENI-1` network namespace. All containers of the task will share this
+network namespace.
+
+Tasks (containers in tasks) launched into these namespaces will be addressable
+by the primary IP Address of the respective secondary network interfaces. For
+example, all containers of the task using ENI `ENI-1` will be addressable within
+the VPC using the IP Address `172.31.13.14`.
+
+The `ecs-eni-br` bridge is used by the containers to communicate with the
```

Comment:
Have you considered any alternative approach? I believe that Task containers could communicate with ECS Agent via unix domain socket. Such approach would be way less complex (no bridge, no IP allocations).